codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerUpgradeCodec.java

```diff
@@ -156,7 +168,7 @@ public void upgradeTo(final ChannelHandlerContext ctx, FullHttpRequest upgradeRe
 
         if (handlers != null) {
             final String name = ctx.pipeline().context(connectionHandler).name();
-            for (int i = handlers.length - 1; i >= 0; i--) {
+            for (int i = handlers.length - 1; i >= firstHandlerIdx; i--) {
                 ctx.pipeline().addAfter(name, null, handlers[i]);
```

Comment:
It isn't clear to me why this is delayed. I'd more expect us to add all the handlers and then call `onHttpServerUpgrade`. Seems this is 15611dadb all over again.

Reply:
That would be fine by me as well... just thought it is good to change as less behavior as possible.

Reply:
@ejona86 should I just do this ?

Reply:
I sort of understood not changing much behavior, but it seems we are working around a bug. I can't think of anything that would go wrong by adding the other handlers, and the extra handlers are really only used in the one case that is broken. So... all signs point to adding all handlers first. But I agree I'm hesitant, mainly because it's unclear why that wasn't done initially.

Reply:
after looking again at  15611da I think you are right and I also can not think of any reason... Let me try to do this and see if we have any breakage.

Reply:
@ejona86 PTAL again